R/textmodel_nb.R

```diff
@@ -331,3 +331,80 @@ colNorm <- function(x) {
 print.predict.textmodel_nb <- function(x, ...) {
     print(unclass(x))
 }
+
+
+#' Compute the information gain of features from a fitted \link{textmodel_nb}
+#' object
+#' 
+#' Computes the information gain of features from a fitted \link{textmodel_nb}
+#' object.  See  
+#' @param model a \link{textmodel_nb} object
+#' @param class_specific  whether to return class specific information gain
+#'   scores or not. Defaults to \code{TRUE}.
+#' @param base the base of the logarithmic function to be used. Defaults to 2.
+#' @references 
+#' Jurafsky, Daniel and James H. Martin.  (2018).
+#' \href{https://web.stanford.edu/~jurafsky/slp3/4.pdf}{\emph{Speech and
+#' Language Processing}}.   Stanford University typescript.  Draft of September
+#' 23.
+#' @keywords textmodel textmodel_nb
+#' @examples 
+#' txt <- c(d1 = "Chinese Beijing Chinese",
+#'          d2 = "Chinese Chinese Shanghai",
+#'          d3 = "Chinese Macao",
+#'          d4 = "Tokyo Japan Chinese",
+#'          d5 = "Chinese Chinese Chinese Tokyo Japan")
+#' trainingset <- dfm(txt, tolower = FALSE)
+#' trainingclass <- factor(c("Y", "Y", "Y", "N", NA), ordered = TRUE)
+#' tmod <- textmodel_nb(trainingset, y = trainingclass, prior = "docfreq")
+#' gain.textmodel_nb(tmod)
+#' gain.textmodel_nb(tmod, class_specific = FALSE)
+#' @export
```

Comment:
we can put this on the man page with `predict.textmodel_nb()`, using `@rdname`.  

We should add this function to `@seealso` for the `textmodel_nb()` function.

Reply:
this is great! were you able to check that the computation is correct?

Reply:
Added `gain` to `@seealso` to the man page for see also `textmodel_nb()`, not too sure what you meant by the `@rdname` part though @kbenoit